examples/webgl2_buffergeometry_attributes_integer.html

```diff
@@ -0,0 +1,171 @@
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+		<link type="text/css" rel="stylesheet" href="main.css">
+	</head>
+	<body>
+
+		<div id="container"></div>
+		<div id="info"><a href="https://threejs.org" target="_blank" rel="noopener">three.js</a> WebGL 2 - buffergeometry - integer attributes</div>
+
+		<script id="vertexShader" type="x-shader/x-vertex">#version 300 es
+			in int textureIndex;
+
+			flat out int vIndex; // "flat" indicates that the value will not be interpolated (required for integer attributes)
+			out vec2 vUv;
+
+			void main()	{
+
+				vIndex = textureIndex;
+				vUv = uv;
+
+				gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );
+
+			}
+		</script>
+
+		<script id="fragmentShader" type="x-shader/x-fragment">#version 300 es
+			flat in int vIndex;
+			in vec2 vUv;
+
+			uniform sampler2D uTextures[ 3 ];
+
+			out vec4 outColor;
+
+			void main()	{
+
+				if ( vIndex == 0 ) outColor = texture( uTextures[ 0 ], vUv );
+				else if ( vIndex == 1 ) outColor = texture( uTextures[ 1 ], vUv );
+				else if ( vIndex == 2 ) outColor = texture( uTextures[ 2 ], vUv );
+
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from '../build/three.module.js';
+
+			var camera, scene, renderer, mesh;
+
+			init();
+			animate();
+
+			function init() {
+
+				camera = new THREE.PerspectiveCamera( 27, window.innerWidth / window.innerHeight, 1, 3500 );
+				camera.position.z = 2500;
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0x050505 );
+				scene.fog = new THREE.Fog( 0x050505, 2000, 3500 );
+
+				// geometry
+
+				var triangles = 10000;
+
+				var geometry = new THREE.BufferGeometry();
+
+				var positions = [];
+				var uvs = [];
+				var textureIndices = [];
+
+				var n = 800, n2 = n / 2; // triangles spread in the cube
+				var d = 50, d2 = d / 2; // individual triangle size
+
+				for ( var i = 0; i < triangles; i ++ ) {
+
+					// positions
+
+					var x = Math.random() * n - n2;
+					var y = Math.random() * n - n2;
+					var z = Math.random() * n - n2;
+
+					var ax = x + Math.random() * d - d2;
+					var ay = y + Math.random() * d - d2;
+					var az = z + Math.random() * d - d2;
+
+					var bx = x + Math.random() * d - d2;
+					var by = y + Math.random() * d - d2;
+					var bz = z + Math.random() * d - d2;
+
+					var cx = x + Math.random() * d - d2;
+					var cy = y + Math.random() * d - d2;
+					var cz = z + Math.random() * d - d2;
+
+					positions.push( ax, ay, az );
+					positions.push( bx, by, bz );
+					positions.push( cx, cy, cz );
+
+					// uvs
+
+					uvs.push( 0, 0 );
+					uvs.push( 0.5, 1 );
+					uvs.push( 1, 0 );
+
+					// texture indices
+
+					var t = i % 3;
+					textureIndices.push( t, t, t );
+
+				}
+
+				geometry.setAttribute( 'position', new THREE.Float32BufferAttribute( positions, 3 ) );
```

Comment:
What if we simplified the example and we just created positions using integers? 🤔

Reply:
We can't use `ShaderMaterial` in this case since it's necessary to overwrite the default `position` attribute (which is `vec3` but we would need `ivec3`). Besides, this use case does not allow to demonstrate the usage of `flat` since the attribute is only needed in the vertex shader.

However, if you still think it's better to simplify the example, I'm okay with the integer position attribute approach.

Reply:
It would be nice if it was visually obvious that the example was working correctly. Maybe a simpler example would make that clear?

Also, do we really need a separate example for each attribute type?

Reply:
> Also, do we really need a separate example for each attribute type?

No, I don't think so. But since integer shader attributes are somewhat special, I think an example is really helpful.

Reply:
> We can't use `ShaderMaterial` in this case since it's necessary to overwrite the default `position` attribute (which is `vec3` but we would need `ivec3`). Besides, this use case does not allow to demonstrate the usage of `flat` since the attribute is only needed in the vertex shader.

That's fair. Okay, lets go with this for now.

> Also, do we really need a separate example for each attribute type?

We could merge examples in the future.